src/dolphin/timeseries.py

```diff
@@ -607,28 +607,33 @@ def correlation_to_variance(correlation: ArrayLike, nlooks: int) -> Array:
 
 def select_reference_point(
     ccl_file_list: Sequence[PathOrStr],
-    amp_dispersion_file: PathOrStr,
+    condition_file: PathOrStr,
     output_dir: Path,
+    condition_func: Callable = np.argmin,
```

Comment:
Should we include the `np.unravel_index()` part in `condition_func`?

It seems a little awkward that right now `condition_func` returns an index into the flattened array, rather than a tuple of indices along each axis of the array. If we want `condition_func` to be generic, maybe a friendlier interface would be for it to return a tuple?

```python
def argmin_index(arr: ArrayLike) -> tuple[int, ...]:
    """Get the index tuple of the minimum value of the array.

    If multiple occurrences of the minimum value exist, returns
    the index of the first such occurrence in the flattened array.

    Parameters
    ----------
    arr : array_like
        The input array.

    Returns
    -------
    tuple of int
        The index of the minimum value.
    """
    return np.unravel_index(arr, np.shape(arr))
```

```suggestion
    condition_func: Callable = argmin_index,
```